AhnChiWook_practice/AhnChiWook_practice/Week1/DangGeunLoginViewController.swift

```diff
@@ -0,0 +1,160 @@
+        $0.setTitle("로그인하기", for: .normal)
+        $0.setTitleColor(.gray300, for: .disabled)
+        $0.setTitleColor(.white, for: .normal)
+        $0.backgroundColor = .gray200
+        $0.addTarget(self, action: #selector(loginButtonDidTap), for: .touchUpInside)
+        $0.layer.cornerRadius = 6
+        $0.isEnabled = false
+    }
+    
+    private lazy var darkThemeButton = UIButton().then {
+        $0.frame = CGRect(x: 30, y: 490, width: 332, height: 58)
+        $0.titleLabel?.font = .pretendard(.bold, size: 18)
+        $0.setTitle("다크 모드", for: .normal)
+        $0.setTitleColor(.white, for: .normal)
+        $0.backgroundColor = .black
+        $0.layer.cornerRadius = 6
+        $0.isSelected = true
+        $0.addTarget(self, action: #selector(darkThemeButtonDidTap), for: .touchUpInside)
+    }
+    
+    
+    // MARK: - Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.view.backgroundColor = .white
+        setLayout()
+    }
+    
+    private func setLayout() {
+        self.view.addSubviews(
+            titleLabel,
+            idTextField,
+            passwordTextField,
+            loginButton,
+            darkThemeButton
+        )
+    }
+    
+    // MARK: - Method
+    
+    private func presentToWelcomeVC() {
+        let vc = DangGeunWelcomeViewController()
+        self.present(vc, animated: true)
+    }
+    
+    private func pushToWelcomeVC() {
+        let welcomeViewController = DangGeunWelcomeViewController()
+        welcomeViewController.name = idTextField.text
+        self.navigationController?.pushViewController(welcomeViewController, animated: true)
+    }
+    
+    @objc
+    private func loginButtonDidTap() {
+        //presentToWelcomeVC()
+        pushToWelcomeVC()
+    }
+    
+    @objc
+    private func updateLoginButtonState() {
+        let isIdValid = (idTextField.text?.isEmpty == false)
+        let isPasswordValid = (passwordTextField.text?.isEmpty == false)
+        let canLogin = isIdValid && isPasswordValid
+        
+        loginButton.isEnabled = canLogin
+        
+        loginButton.backgroundColor = canLogin ? .dangGeun : .gray200
+    }
+    
+    @objc
+    private func darkThemeButtonDidTap() {
+        if darkThemeButton.isSelected {
+            self.view.backgroundColor = .black
+            self.titleLabel.textColor = .white
+            darkThemeButton.setTitle("라이트 모드", for: .normal)
+            darkThemeButton.backgroundColor = .white
+            darkThemeButton.setTitleColor(.black, for: .normal)
+            darkThemeButton.isSelected = false
+        }
+        else {
+            self.view.backgroundColor = .white
+            self.titleLabel.textColor = .black
+            darkThemeButton.setTitle("다크 모드", for: .normal)
+            darkThemeButton.backgroundColor = .black
+            darkThemeButton.setTitleColor(.white, for: .normal)
+            darkThemeButton.isSelected = true
+        }
+    }
```

Comment:
위에 삼항 연산자 쓴거 처럼 이 함수도 적용을 한다면, 이렇게 깔끔하게도 가능해요 ㅎㅎ

```suggestion
    
    @objc
    private func darkThemeButtonDidTap() {
        darkThemeButton.isSelected.toggle()
        let isDarkMode = darkThemeButton.isSelected
    
        view.backgroundColor = isDarkMode ? .black : .white
        titleLabel.textColor = isDarkMode ? .white : .black
    
        darkThemeButton.setTitle(isDarkMode ? "라이트 모드" : "다크 모드", for: .normal)
        darkThemeButton.backgroundColor = isDarkMode ? .white : .black
        darkThemeButton.setTitleColor(isDarkMode ? .black : .white, for: .normal)
    }
```